src/basics.jl

```diff
@@ -504,7 +504,24 @@ function rule(pos, theta=0.0;
     return ruledline
 end
 
-saved_colors = Tuple{Float64, Float64, Float64, Float64}[]
+let SAVED_COLORS_STACK = Ref{Dict{Int,Array{Tuple{Float64, Float64, Float64, Float64},1}}}(Dict(0 => Array{Tuple{Float64, Float64, Float64, Float64},1}()))
+    global SAVED_COLORS
+    function SAVED_COLORS()
```

Comment:
I suppose these have to be UpperCase...  :( It gets a bit difficult to read all the upper-case names later on. 

BlueStyle says:
>NOTE: Functions that are only intended for internal use should be marked with a leading underscore (e.g., _internal_utility_function(X, y)). Although it should be much less common, the same naming convention can be used for internal types and constants as well (e.g., _MyInternalType, _MY_CONSTANT). Marking a function as internal or private lets other people know that they shouldn't expect any kind of API stability from that functionality.

Perhaps we should start doing that now, since there are now quite a few global things floating around...